fastestimator/trace/io/model_eval.py

```diff
@@ -0,0 +1,451 @@
+                            self._document_instance_table(tests=instance_fail_tests, with_ID=self.data_id)
+
+        if instance_pass_tests or aggregate_pass_tests:
+            with self.doc.create(Section("Passed Tests")):
+                if aggregate_pass_tests:
+                    with self.doc.create(Subsection("Passed Aggregate Tests")):
+                        with self.doc.create(Table(position='H')) as table:
+                            table.append(NoEscape(r'\refstepcounter{table}'))
+                            self._document_aggregate_table(tests=aggregate_pass_tests)
+                if instance_pass_tests:
+                    with self.doc.create(Subsection("Passed Per-Instance Tests")):
+                        with self.doc.create(Table(position='H')) as table:
+                            table.append(NoEscape(r'\refstepcounter{table}'))
+                            self._document_instance_table(tests=instance_pass_tests, with_ID=self.data_id)
+
+    def _document_summary_table(self, pass_num: int, fail_num: int) -> None:
+        """Document a summary table.
+
+        Args:
+            pass_num: Total number of passed tests.
+            fail_num: Total number of failed tests.
+        """
+        with self.doc.create(Tabularx('|Y|Y|Y|', booktabs=True)) as tabular:
+            package = Package('seqsplit')
+            if package not in tabular.packages:
+                tabular.packages.append(package)
+
+            # add table heading
+            tabular.add_row(("Total test #", "Total pass #", "Total fail #"), strict=False)
+            tabular.add_hline()
+
+            tabular.add_row((pass_num + fail_num, pass_num, fail_num), strict=False)
+
+    def _document_instance_table(self, tests: List[dict], with_ID: bool):
+        """Document a result table of per-instance tests.
+
+        Args:
+            tests: List of corresponding test dictionary to make a table.
+            with_ID: Whether the test information includes data ID.
+        """
+        if with_ID:
+            table_spec = '|c|X|c|c|X|'
+        else:
+            table_spec = '|c|X|c|c|'
+
+        with self.doc.create(Tabularx(table_spec, booktabs=True)) as tabular:
+            package = Package('seqsplit')
+            if package not in tabular.packages:
+                tabular.packages.append(package)
+
+            # add table heading
+            row_cells = [
+                MultiColumn(size=1, align='|c|', data="Test ID"),
+                MultiColumn(size=1, align='c|', data="Test description"),
+                MultiColumn(size=1, align='c|', data="Pass threshold"),
+                MultiColumn(size=1, align='c|', data="Failure count")
+            ]
+
+            if with_ID:
+                row_cells.append(MultiColumn(size=1, align='c|', data="Failure data instance ID"))
+
+            tabular.add_row(row_cells)
+            tabular.add_hline()
+
+            for idx, test in enumerate(tests, 1):
+                if idx > 1:
```

Comment:
you could remove this if check if you remove the add_hline on line 320 right?

Reply:
and in fact, then you can remove idx from the loop entirely and drop the enumerate right?